### DISCUSSION TITLE
How to access array elements in json

### DISCUSSION THREAD

**Discussion by HimanshuGohel** [nlohmann/json#4089] [Q&A]
example:-

{
     "info":[
                 {
                     "Person":"xyz",
                     "city":"abc"
                 },
                 {
                     "Person":"pqr",
                     "city":"mno"
                 }
     ]

}

**Comment by nlohmann** [comment]
Parse the text into a json object, then index into the array "info" and each object's fields: json j = json::parse(text); std::string person = j["info"][0]["Person"]; std::string city = j["info"][0]["city"]; . To process every entry, iterate: for (auto& entry : j["info"]) { std::string p = entry["Person"]; std::string c = entry["city"]; }. See docs/mkdocs/docs/features/element_access.md and iterators.md for the general access patterns.

---
*This reply was drafted by [Claude Code](https://claude.com/claude-code) on behalf of @nlohmann, as part of a review of open discussions.*
